src/plan/plan_constraints.rs

```diff
@@ -45,6 +45,10 @@ pub struct PlanConstraints {
     /// `MutatorConfig::prepare_func`).  Those plans can set this to `false` so that the
     /// `PrepareMutator` work packets will not be created at all.
     pub needs_prepare_mutator: bool,
+    /// Should a policy unlog newly allocated objects?
+    pub unlog_allocated_object: bool,
```

Comment:
I think this field needs to be defined per space, rather than per plan. In our generational plans, this value should be `false` for the nursery space, and any space that allows mixed generation (such as LOS, or sticky immix's immix space), and the value should be `true` for other spaces.

Reply:
Hm. It should also be `false` for LOS since it has a nursery. What spaces do we directly allocate into that would need this for generational GCs? Immortal space, I guess. Because LOS and nursery space should be the main spaces where new objects are allocated into. Again, I'm only talking about generational GCs. I'm aware that SATB requires this.

Reply:
> What spaces do we directly allocate into that would need this for generational GCs?

Immortal, non moving, code, VM space (we don't allocate into VM space, but we ask the binding to call our post_alloc after their allocation), etc. Also if we allow pretenure allocation, we may directly allocate into a mature space and also unlog the objects.